### PR TITLE
Bryn/otel config merge

### PR DESCRIPTION
**Pull request by BrynCooke**

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/metrics/otlp.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        Self {
+            exporter: Some(exporter),
+        }
+    }
+}
+
+impl From<HttpExporterBuilder> for MetricExporterBuilder {
+    // This has to exist because Http is not currently supported for metrics export
+    // https://github.com/open-telemetry/opentelemetry-rust/issues/772
+    fn from(_exporter: HttpExporterBuilder) -> Self {
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/metrics/otlp.rs:27]: Could you add a TODO here ?

File: apollo-router/src/plugins/telemetry/otlp.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    Grpc,
+    Http,
+}
+
+impl Default for Protocol {
+    fn default() -> Self {
+        vec![Protocol::Grpc, Protocol::Http]
+            .get(0)
+            .expect("at least one feature must be present for otel, qed")
+            .clone()
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/otlp.rs:165]:
```suggestion
        Protocol::Grpc
```

File: apollo-router/src/plugins/telemetry/otlp.rs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    }
+}
+
+mod header_map_serde {
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/otlp.rs:169]: I think it should be `metadata_map_serde`

File: docs/source/configuration/tracing.mdx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+The `trace_config` section contains common configuration that will be used in the for all exporters. It is optional and will fall back on env variables as specified by the [OpenTelemetry spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md) if `service_name` is not set.
+
+```yaml title="router.yaml"
+telemetry:
+  tracing:
+    trace_config:
+      service_name: "${MY_ENV_VARIABLE:router}"
+      service_namespace: "apollo"
+      # Optional. Either a float between 0 and 1 or 'AlwaysOn' or 'AlwaysOff'
+      sampler: AlwaysOn
```
Comment (bnjjj) [inline docs/source/configuration/tracing.mdx:37]: I wonder if we should stay consistent and have snake_case everywhere ?

**Comment by BrynCooke** [comment]
I will fold all changes in to #782

File: docs/source/configuration/tracing.mdx (hunk 4 above)
Comment (BrynCooke) [reply docs/source/configuration/tracing.mdx:37]: Yeah, I missed this first time round. But this is now fixed.

File: apollo-router/src/plugins/telemetry/otlp.rs (hunk 3 above)
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/otlp.rs:169]: Fixed

File: apollo-router/src/plugins/telemetry/otlp.rs (hunk 2 above)
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/otlp.rs:165]: Fixed

File: apollo-router/src/plugins/telemetry/metrics/otlp.rs (hunk 1 above)
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/metrics/otlp.rs:27]: Fixed

**Pull request closed without merging**